grammar/it/src/rules.rs

```diff
@@ -0,0 +1,513 @@
+                      |text_match| IntegerValue::new(text_match.group(0).parse()?));
+    b.rule_1_terminal("integer with thousands separator .",
+                      b.reg(r#"(\d{1,3}(\.\d\d\d){1,5})"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(".", "");
+                          let value: i64 = reformatted_string.parse()?;
+                          IntegerValue::new(value)
+                      });
+    b.rule_1_terminal("decimal number",
+                      b.reg(r#"(\d*,\d+)"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(",", ".");
+                          let value: f32 = reformatted_string.parse()?;
+                          FloatValue::new(value)
+                      });
+    b.rule_3("number dot number",
+             number_check!(|number: &NumberValue| !number.prefixed()),
+             b.reg(r#"punto"#)?,
+             number_check!(|number: &NumberValue| !number.suffixed()),
+             |a, _, b| {
+                 Ok(FloatValue {
+                     value: b.value().value() * 0.1 + a.value().value(),
+                     ..FloatValue::default()
+                 })
+             });
+    b.rule_1_terminal("decimal with thousands separator",
+                      b.reg(r#"(\d+(\.\d\d\d)+,\d+)"#)?,
+                      |text_match| {
+                          let reformatted_string = text_match.group(1).replace(".", "").replace(",", ".");
+                          let value: f32 = reformatted_string.parse()?;
+                          FloatValue::new(value)
+                      });
+    b.rule_2("numbers prefix with -, negative or minus",
+             b.reg(r#"-|meno"#)?,
+             number_check!(|number: &NumberValue| !number.prefixed()),
+             |_, a| -> RuleResult<NumberValue> {
+                 Ok(match a.value().clone() {
+                     // checked
+                     NumberValue::Integer(integer) => {
+                         IntegerValue {
+                             value: integer.value * -1,
+                             prefixed: true,
+                             ..integer
+                         }
+                             .into()
+                     }
+                     NumberValue::Float(float) => {
+                         FloatValue {
+                             value: float.value * -1.0,
+                             prefixed: true,
+                             ..float
+                         }
+                             .into()
+                     }
+                 })
+             });
+    b.rule_1_terminal("ordinals (primo..10)",
+                      b.reg(r#"((?:prim|second|terz|quart|quint|sest|settim|ottav|non|decim)[oiae])"#)?,
+                      |text_match| {
+                          let value = match text_match.group(1).as_ref() {
+                              "primo" => 1,
```

Comment:
If you replace `((?:prim|second|terz|quart|quint|sest|settim|ottav|non|decim)[oiae])` by `(prim|second|terz|quart|quint|sest|settim|ottav|non|decim)[oiae]` you can reduce the number of cases in the match pattern.

Reply:
OK, will change - we don't need to have the whole token captured for the output then, i.e. radical + [oiae]?